ETLParser.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+
+# Parser for PEDSnet ETL conventions document
+
+import json
+import os.path
+import re
+import sys
+
+_h1 = re.compile(r'^\#[^\#]+')
+_h2 = re.compile(r'^\#\#[^\#]+')
+
+
+class Document():
+    def __init__(self, path):
+        self.dict = {}
+        self.dict['content'] = ''
+        self.dict['tables'] = {}
+
+        with open(path, "r") as f:
+            self.lines = f.readlines()
+
+        self.get_content()
+
+        while self.pos < len(self.lines):
+            self.get_next_table()
+
+    def add_content(self, content):
+        self.dict['content'] = content
+
+    def add_table(self, table, content=''):
+        self.dict['tables'][table] = {}
+        self.dict['tables'][table]['content'] = content
+        self.dict['tables'][table]['fields'] = {}
+
+    def add_table_content(self, table, content):
+        self.dict['tables'][table]['content'] = content
+
+    def add_field(self, table, field,
+                  required, data_type, desc, conv):
+        self.dict['tables'][table]['fields'][field] = {}
+
+        self.dict['tables'][table]['fields'][field]['required'] = required
+        self.dict['tables'][table]['fields'][field]['data_type'] = data_type
+        self.dict['tables'][table]['fields'][field]['description'] = desc
+        self.dict['tables'][table]['fields'][field]['etl_conventions'] = conv
+
+    def __str__(self):
+        return json.dumps(self.dict, indent=4)
+
+    def get_content(self):
+        self.pos = 0
+
+        # Skip everything up to and including the h1 element (the title).
+        while self.pos < len(self.lines):
+            line = self.lines[self.pos]
+            self.pos = self.pos + 1
+            if _h1.match(line):
+                break
+
+        # Everything up to the first h2 element is model-level content.
+        content = ''
```

Comment:
Not that it would have a great effect here, but as a general performance consideration, it is more efficient to append strings to a list then join them at the end.